src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.IllegalStateException;
+import java.io.IOException;
+
+/**
+ * This servlet handles placeguide's data.
+ */
+@WebServlet("/place-guide-data")
+public class PlaceGuideServlet extends HttpServlet {
+  public static final String ALL = "all";
+  public static final String CREATED_ALL = "created-all";
+  public static final String CREATED_PUBLIC = "created-public";
+  public static final String CREATED_PRIVATE = "created-private";
+  public static final String NAME_INPUT = "name";
+  public static final String AUDIO_KEY_INPUT = "audioKey";
+  public static final String PLACE_ID_INPUT = "placeId";
+  public static final String IS_PUBLIC_INPUT = "isPublic";
+  public static final String IS_PUBLIC_INPUT_VALUE = "public";
+  public static final String LATITUDE_INPUT = "latitude";
+  public static final String LONGITUDE_INPUT = "longitude";
+  public static final String DESCRIPTION_INPUT = "description";
+  public static final String LENGTH_INPUT = "length";
+  public static final String IMG_KEY_INPUT = "imgKey";
+  public static final String PLACE_GUIDE_TYPE_PARAMETER = "placeGuideType";
+
+  private final PlaceGuideRepository placeGuideRepository = PlaceGuideRepositoryFactory
+                                                .getPlaceGuideRepository(RepositoryType.DATASTORE);
+  private final String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+
+  /**
+   * Saves the recently submitted place guide data.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PlaceGuide placeGuide = getPlaceGuideFromRequest(request);
+    placeGuideRepository.savePlaceGuide(placeGuide);
+    response.sendRedirect("/createPlaceGuide.html");
+  }
+
+  /**
+   * Returns the data of the placeguide(s) asked by the user who is currently logged in.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String placeGuideType = request.getParameter(PLACE_GUIDE_TYPE_PARAMETER);
+    List<PlaceGuide> placeGuides;
+    switch(placeGuideType) {
+      case ALL:
+        placeGuides = placeGuideRepository.getAllPlaceGuides();
+        break;
+      case CREATED_ALL:
+        placeGuides = placeGuideRepository.getCreatedPlaceGuides(userId);
+        break;
+      case CREATED_PUBLIC:
+        placeGuides = placeGuideRepository.getCreatedPublicPlaceGuides(userId);
+        break;
+      case CREATED_PRIVATE:
+        placeGuides = placeGuideRepository.getCreatedPrivatePlaceGuides(userId);
+        break;
+      default:
+        throw new IllegalStateException("Place Guide type does not exist!");
+    }
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJsonUsingGson(placeGuides));
+  }
+
+  private PlaceGuide getPlaceGuideFromRequest(HttpServletRequest request) {
```

Comment:
Since this servlet should also handle updating an existing place guide's data, we should be able to get a place guide's ID from the request as well.

Reply:
set the id if the id parameter is not empty.